src/GafferDelight/IECoreDelightPreview/Renderer.cpp

```diff
@@ -0,0 +1,1290 @@
+			h.release();
+			return *this;
+		}
+
+		NSIContext_t context() const
+		{
+			return m_context;
+		}
+
+		const char *name() const
+		{
+			return m_name.c_str();
+		}
+
+		Ownership ownership() const
+		{
+			return m_ownership;
+		}
+
+		void reset()
+		{
+			if( m_ownership == Owned && m_context != NSI_BAD_CONTEXT )
+			{
+				NSIDelete( m_context, m_name.c_str(), 0, nullptr );
+			}
+			release();
+		}
+
+	private :
+
+		void release()
+		{
+			m_context = NSI_BAD_CONTEXT;
+			m_name = "";
+			m_ownership = Unowned;
+		}
+
+		NSIContext_t m_context;
+		std::string m_name;
+		Ownership m_ownership;
+
+};
+
+typedef std::shared_ptr<DelightHandle> DelightHandleSharedPtr;
+typedef std::weak_ptr<DelightHandle> DelightHandleWeakPtr;
+
+} // namespace
+
+//////////////////////////////////////////////////////////////////////////
+// DelightOutput
+//////////////////////////////////////////////////////////////////////////
+
+namespace
+{
+
+class DelightOutput : public IECore::RefCounted
+{
+
+	public :
+
+		DelightOutput( NSIContext_t context, const std::string &name, const IECoreScenePreview::Renderer::Output *output, DelightHandle::Ownership ownership )
+			:	m_context( context )
+		{
+			const char *typePtr = output->getType().c_str();
+			const char *namePtr = output->getName().c_str();
+
+			ParameterList driverParams( output->parameters() );
+			driverParams.add( { "drivername", &typePtr, NSITypeString, 0, 1 } );
+			driverParams.add( { "imagefilename", &namePtr, NSITypeString, 0, 1 } );
+
+			m_driverHandle = DelightHandle( context, "outputDriver:" + name, ownership, "outputdriver", driverParams );
+
+			const char *ciPtr = "Ci";
+			const char *shaderPtr = "shader";
+			const char *rgbaPtr = "rgba";
+			const char *linearPtr = "linear";
+			const char *layerTypePtr = "color";
+			const char *scalarFormatPtr = scalarFormat( output );
+			const int one = 1;
+			ParameterList layerParams = {
+				{ "variablename", &ciPtr, NSITypeString, 0, 1 },
+				{ "variablesource", &shaderPtr, NSITypeString, 0, 1 },
+				{ "layertype", &layerTypePtr, NSITypeString, 0, 1 },
+				{ "layername", &rgbaPtr, NSITypeString, 0, 1 },
+				{ "colorprofile", &linearPtr, NSITypeString, 0, 1 },
+				{ "withalpha", &one, NSITypeInteger, 0, 1 },
+				{ "scalarformat", &scalarFormatPtr, NSITypeString, 0, 1 }
+			};
+
+			m_layerHandle = DelightHandle( context, "outputLayer:" + name, ownership, "outputlayer", layerParams );
+
+			NSIConnect(
+				m_context,
+				m_driverHandle.name(), "",
+				m_layerHandle.name(), "outputdrivers",
+				0, nullptr
+			);
+		}
+
+		const DelightHandle &layerHandle() const
+		{
+			return m_layerHandle;
+		}
+
+	private :
+
+		const char *scalarFormat( const IECoreScenePreview::Renderer::Output *output ) const
+		{
+			// Map old-school "quantize" setting to scalarformat. Maybe
+			// we should have a standard more suitable for mapping to modern
+			// renderers and display drivers? How would we request half outputs
+			// for instance?
```

Comment:
Strongly agree that "quantize" is not a good way to represent this.

Reply:
Once this is merged and we're only using the new backends I'd like to take a look at improving this situation. Any thoughts welcome...

Reply:
My first thought would just be to expose something like "scalarFormat", with an enum value?

Reply:
Yeah, summink like that would do, as long as we're happy with one format per file. Might be worth discussing whether or not we want to adopt separate parameters per aov and multiple aovs per file though. Anyways, later...